kernelci/config/base.py

```diff
@@ -171,10 +205,25 @@ class FilterFactory(YAMLObject):
     def from_yaml(cls, filter_params):
         """Iterate through the YAML filters and return Filter objects."""
         filter_list = []
+        filters = {}
+
         for f in filter_params:
             for filter_type, items in f.items():
-                filter_cls = cls._classes[filter_type]
-                filter_list.append(filter_cls(items))
+                if filter_type not in filters:
+                    filter_cls = cls._classes[filter_type]
+                    filter_instance = filter_cls(items)
+                    filters[filter_type] = [filter_instance]
+                    filter_list.append(filter_instance)
+                else:
+                    for g in filters[filter_type]:
+                        if g.combine(items):
+                            break
+                    else:
+                        filter_cls = cls._classes[filter_type]
+                        filter_instance = filter_cls(items)
+                        filters[filter_type].append(filter_instance)
+                        filter_list.append(filter_instance)
```

Comment:
This block looks very similar to lines 212-216. Maybe we could extract it to a helper function/method?